cmd/cloud_sql_proxy/cloud_sql_proxy.go

```diff
@@ -373,6 +377,24 @@ func gcloudProject() []string {
 	return []string{data.Core.Project}
 }
 
+func setupTermDelay() {
+	terminated := make(chan bool, 0)
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGTERM)
+
+	go func() {
+		<-terminated
+		time.Sleep(*termDelay)
```

Comment:
Sorry, while I'm hoping for unicorns: we should also call `close(connSrc)` here as well, so that the proxy stops accepting new connections at this point.